ORLib/src/main/java/io/openremote/orlib/ui/OrMainActivity.kt

```diff
@@ -797,34 +799,26 @@ open class OrMainActivity : Activity() {
         @Throws(JSONException::class)
         private fun handleStorageProviderMessage(data: JSONObject) {
             val action = data.getString("action")
+            if (secureStorageProvider == null) {
+                secureStorageProvider = SecureStorageProvider(activity)
+            }
             when {
                 action.equals("PROVIDER_INIT", ignoreCase = true) -> {
-                    val response: MutableMap<String, Any> = HashMap()
-                    response["action"] = "PROVIDER_INIT"
-                    response["provider"] = "storage"
-                    response["version"] = "1.0.0"
-                    response["enabled"] = true
-                    response["requiresPermission"] = false
-                    response["hasPermission"] = true
-                    response["success"] = true
+                    val response = secureStorageProvider!!.initialize()
                     notifyClient(response)
                 }
 
                 action.equals("PROVIDER_ENABLE", ignoreCase = true) -> {
                     // Doesn't require enabling but just in case it gets called lets return a valid response
-                    val response: MutableMap<String, Any> = HashMap()
-                    response["action"] = "PROVIDER_ENABLE"
-                    response["provider"] = "storage"
-                    response["hasPermission"] = true
-                    response["success"] = true
+                    val response = secureStorageProvider!!.enable()
```

Comment:
Does this provider need enabling now as the JS code expects this provider to be auto enabled

Reply:
I do see it being called. I'm not sure what will happen when I remove this part. Will the console leader continue or stop because it waits for the enable response?

Reply:
If `storage` is included in the providers list then indeed init will be called but the JS code uses the storage provider before any provider init logic so it is expected to always be ready.

Android and iOS code should just auto initialise this provider on app creation and any call to storage provider init should jsut do nothing.

Reply:
I didn't realise the `initialise()` and `enable()` methods of the SecureStorageProvider only returned the JSON response, they don't actually do anything so that is fine, and can stay...I will revert that.